remove_debugging_statements.yaml

```diff
@@ -0,0 +1,47 @@
+# These optional rules flag or remove various debugging statements.
+#
+# All rules in this file have the tag `no-debug`.
+#
+
+rules:
+- id: remove-breakpoint
+  description: Remove breakpoints from production code
+  explanation: Breakpoints shouldn't be used in production code.
+  pattern: breakpoint()
+  replacement: ''
+  tests:
+  - match: breakpoint()
+  - no-match: breakpoint("sth")
+  tags:
+  - no-debug
+
+- id: remove-pdb-set_trace
+  description: Remove `pdb.set_trace` calls from production code
+  explanation: Breakpoints shouldn't be used in production code.
+  pattern: pdb.set_trace()
```

Comment:
Just a note here: [`pdb.set_trace`](https://docs.python.org/3/library/pdb.html#pdb.set_trace) also accepts an optional `header` argument, so that the most general version of this rule would be `pdb.set_trace(...)`.
However, care must be taken to ensure that the `header` argument does not write variables, which we cannot check for at the moment...
Could you please add a note for that, even if it is just a to-do item for us to review later?

Reply:
On second thought, I think that we should _always_ flag `pdb.set_trace(...)` - and then it's up to the user to handle side-effects or to directly accept Sourcery's suggestion to remove that call altogether.
So I propose that we generalize this pattern to
```suggestion
  pattern: pdb.set_trace(...)
```